src/file_part.h

```diff
@@ -42,10 +42,10 @@ class FilePart {
         m_filename(filename),
         m_fhandle(std::make_shared<FS::FD>(FS::openFile(filename))),
         m_size(m_fhandle->getSize()) {}
+#ifndef _WIN32
     FilePart(int fd) :
-        m_filename(""),
-        m_fhandle(std::make_shared<FS::FD>(fd)),
-        m_size(m_fhandle->getSize()) {}
+        FilePart(getFilePathFromFD(fd)) {}
```

Comment:
We should not resolve the symbolic link.
On linux, it is possible to rename a file or even delete it while it is opened.
The the real path may be wrong when we try to reopen it.
Linux should keep the symbolic link valid in case of rename (I don't know in case of deletion).
We should simply try to open the symlink `/dev/fd/<number>`.

Reply:
> Linux should keep the symbolic link valid in case of rename (I don't know in case of deletion).

I checked that the symbolic links in /proc/*pid*/fd are indeed updated when a file is renamed. When a file is deleted the symbolic link becomes invalid (with ` (deleted)` appended to that last known filename) but still usable until the process terminates.

Reply:
Done